### DISCUSSION TITLE
[New Feature] Event for placing a building at specified location

### DISCUSSION THREAD

**Discussion by Rampastring** [Vinifera-Developers/Vinifera#1321] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

A new trigger event for checking whether a building has been built into a specific location. The location could be denoted by a celltag or a waypoint, whichever is easier to implement.

Imagine a mission where the player(s) is/are tasked to build a specific structure into a specific location, say a powerful air defense on a strategic hill.

With the current events, you're able to check whether a player has built a building, but not where it was placed.

Basically it could work like the existing "Building exists" event, but be also linked to a location on the map.

For the check, it'd be good if any of the building's foundation cells existing on the linked cell counted as the building existing on the position.

**Comment by JoyfulShush** [comment]
This is a neat idea, however I would also want the event to introduce a distance value from the waypoint/cell/celltag, where the structure would be in up to X cells away (in any of its foundation cells) from the origin.

The reason for it is because players hate being told exactly where to build a building - see TR for example which forced players to deploy their MCV in a specific location. So just noticing that the building was built "in the vicinity" should suffice.
